internal/pkg/skuba/kubernetes/resources.go

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package kubernetes
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// DoesResourceExist check the given error from a client-go API call and returns
+// whether the resource exists. If no error was given, it will return true, if
+// the given error was a `IsNotFound` it will return false; otherwise it will
+// return an error
+func DoesResourceExist(err error) (bool, error) {
```

Comment:
can we name this a bit more clear, that it's about error resources?

Reply:
I have been trying to come up with a better name but I couldn't find one, given the transformation this function performs, do you have a better suggestion?

Reply:
i'm not very creative here either, but it would help if we know that it requires an error to be passed, maybe `DoesResourceExistWithError`?

my confusion is that the name `DoesResourceExist` implies passing a resource, but maybe it's just me